### PR TITLE
Documentation: Azure Functions Deployment

### PR DESCRIPTION
**Pull request by SameerDoshi**
Adding additional documentation on accessing request headers when dep…loying Apollo Server in an Azure function.

Big picture: Running Apollo in an Azure Function where Apollo access protected backend APIs a user might want to pass a user provided token to the backend API.

### CODE REVIEW

File: packages/apollo-server-azure-functions/README.md [hunk 1]
```diff
@@ -138,6 +138,26 @@ The options correspond to the [express cors configuration](https://github.com/ex
 * `credentials`: boolean
 * `maxAge`: number
 
+### Request Body / Token
+By default Azure Functions will pass the request object as req.  Pass this into the context of the ApolloServer constructor to access request contents.  For example for passing an auth token to a backend.
```
Comment (glasser) [inline packages/apollo-server-azure-functions/README.md:142]: req is just the name you're choosing for an argument here, FWIW.

File: packages/apollo-server-azure-functions/README.md [hunk 2]
```diff
@@ -138,6 +138,26 @@ The options correspond to the [express cors configuration](https://github.com/ex
...
+
+```js
+const server = new ApolloServer(
+    { typeDefs,       
+        resolvers,
+        context: (req) => {
+          const authHeader = req.context.bindings.req.headers.authorization || '';
```
Comment (glasser) [inline packages/apollo-server-azure-functions/README.md:149]: Is that actually the best way to get a header? I see at https://www.apollographql.com/docs/apollo-server/api/apollo-server/#middleware-specific-context-fields that we pass `context` and also `request`. Wouldn't `req.request.headers` be simpler?

File: packages/apollo-server-azure-functions/README.md [hunk 3]
```diff
@@ -138,6 +138,26 @@ The options correspond to the [express cors configuration](https://github.com/ex
...
+          if (authHeader === '') throw new AuthenticationError('please send       Bearer token with request');
```
Comment (glasser) [inline packages/apollo-server-azure-functions/README.md:150]: weird spacing in the error message

File: packages/apollo-server-azure-functions/README.md [hunk 4]
```diff
@@ -138,6 +138,26 @@ The options correspond to the [express cors configuration](https://github.com/ex
...
+          return {
+            token: authHeader,
+          }
+        },
+```
+
+### Optional, helpful packages
+* azure-function-log-intercept - console.* messages appear in teh functions log.
```
Comment (glasser) [inline packages/apollo-server-azure-functions/README.md:158]: Not sure how helpful this is without an example of how to use it. Also typo "teh" vs "the".

**Review by glasser — CHANGES_REQUESTED** [review]
For what it's worth, I'd suggest we focus more on our docs site and less on the individual package READMEs instead of having two separate places to look for information.

https://www.apollographql.com/docs/apollo-server/deployment/azure-functions/ is the docs page, with source at `docs/source/deployment/azure-functions.md`. The [Lambda docs page](https://www.apollographql.com/docs/apollo-server/deployment/lambda/#getting-request-info) has a nice section on this sort of thing that could be copied and adapted into the Azure section.

**Comment by trevor-scheer** [comment]
Going to close this since it's no longer mergeable. I've captured it in a [separate issue here](https://github.com/apollographql/apollo-server/issues/7056#issuecomment-1283141566) which is concerned with providing getting started guides for serverless applications.

**Pull request closed without merging**